packages/block-library/src/embed/edit.native.js

```diff
@@ -148,7 +149,7 @@ const EmbedEdit = ( props ) => {
 
 	const blockProps = useBlockProps();
 
-	if ( fetching ) {
+	if ( fetching || ( attributesUrl && ! embedPreviewResponse ) ) {
```

Comment:
I noticed that on the web version this glitch is also happening although it's less visible.

From what I checked, looks like the problem is caused by the state update after setting the URL, it requests the embed preview but `isRequestingEmbedPreview` is returning `false`. Then the embed preview request triggers another state update where `isRequestingEmbedPreview` finally returns `true`.

I'm wondering if we should include this case in the calculation of the `fetching` variable within the `useSelect` block instead of here, as looks like it's part of the logic, wdyt?

Reply:
Sounds good to me 👍 Both variables are already present in the hook, I've moved the logic there as well.